### PR TITLE
Feature/1027 get all entity types

### PR DESCRIPTION
**Pull request by kzangeli**
Implements #1027 

GET /v2/type

This request has two URI params:

* options=count
* options=values

*count* is implemented, but as **options** itself is not implemented yet (we need to save a vector for all the options in *options*, and a lookup function), the implementation isd based on a simple comparison:

    uriParamV["options"] == "count"

Good enough for now, until **options** itself is implemented.

**options=values** is not implemented.

Two issues created: #1169 and #1170

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -25,6 +25,7 @@ Add:  URI parameter 'q' for GET /v2/entities (Issue #967)
 Add:  PUT /v2/entities/<entityId>/attrs/<attrName>/value (Issue #999)
 Add:  DELETE /v2/entities/<entityId>/attrs/<attrName> (Issue #993)
 Add:  PUT /v2/entities/<entityId>/attrs/<attrName> (Issue #991)
-Add:  GET /v2/types/<type> (Issue #1028)
+Add:  GET /v2/type/<type> (Issue #1028)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: It should be "/v2/types" in plural, as "/v2/entities", "/v2/registrations", "/v2/subscriptions". All the operations in the NGSIv2 use  plural in the URL (I think is a common idiom in REST).

I think the .apib has a typo in the retrieve entity type operation. It should be also fixed.

File: CHANGES_NEXT_RELEASE [hunk 2]
```diff
@@ -25,6 +25,7 @@ Add:  URI parameter 'q' for GET /v2/entities (Issue #967)
...
 Add:  APPEND_STRICT action type for POST /v1/updateContext operation
 Add:  REPLACE action type for POST /v1/updateContext operation
+Add:  GET /v2/type (Issue #1027)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: Same here.

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: In https://docs.google.com/spreadsheets/d/1f4m624nmO3jRjNalGE11lFLQixnfCENMV6dc54wUCCg/edit#gid=50130961 the URl are rigth.

File: src/app/contextBroker/contextBroker.cpp [hunk 3]
```diff
@@ -371,6 +372,9 @@ PaArgument paArgs[] =
 #define IENTATTRVAL_COMPS_V2    6, { "v2", "entities", "*", "attrs", "*", "value" }
 #define IENTATTRVAL_COMPS_WORD  ""
 
+#define ETT                     EntityAllTypesRequest
+#define ETT_COMPS_V2            2, { "v2", "type" }
```
Comment (fgalan) [inline src/app/contextBroker/contextBroker.cpp]: type -> types

(We shoul also check the route for /v2/types/<type>, although it was developed in a previous PR).

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: Very glad to hear that.
I never liked the /v2/type path ...

File: src/app/contextBroker/contextBroker.cpp (hunk 3 above)
Comment (kzangeli) [inline src/app/contextBroker/contextBroker.cpp]: Fixed in 2818806

File: CHANGES_NEXT_RELEASE (hunk 2 above)
Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: Fixed in 2818806

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: Fixed in 2818806

File: CHANGES_NEXT_RELEASE (hunk 2 above)
Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: CNR fixed in 8d5467e

**Comment by fgalan** [comment]
LGTM

**Pull request merged**